lib/rules/no-dupe-keys.js

```diff
@@ -39,24 +42,43 @@ module.exports = {
     const options = context.options[0] || {}
     const groups = new Set([...GROUP_NAMES, ...(options.groups || [])])
 
-    return utils.executeOnVue(context, (obj) => {
-      /** @type {Set<string>} */
-      const usedNames = new Set()
-      const properties = utils.iterateProperties(obj, groups)
-
-      for (const o of properties) {
-        if (usedNames.has(o.name)) {
-          context.report({
-            node: o.node,
-            message: "Duplicated key '{{name}}'.",
-            data: {
-              name: o.name
-            }
-          })
+    return utils.compositingVisitors(
+      utils.executeOnVue(context, (obj) => {
+        const properties = utils.iterateProperties(obj, groups)
+        /** @type {Set<string>} */
+        const usedNames = new Set()
+        for (const o of properties) {
+          if (usedNames.has(o.name)) {
+            context.report({
+              node: o.node,
+              message: "Duplicated key '{{name}}'.",
+              data: {
+                name: o.name
+              }
+            })
+          }
+
+          usedNames.add(o.name)
         }
+      }),
+      utils.defineScriptSetupVisitor(context, {
+        onDefinePropsEnter(node, props) {
+          for (const prop of props) {
+            if (!prop.propName) continue
 
-        usedNames.add(o.name)
-      }
-    })
+            const variable = findVariable(context.getScope(), prop.propName)
+            if (!variable) continue
```

Comment:
Thank you for opening this PR!
It looks mostly good to me, but I have one concern.
Presumably the current implementation warns against declaring a prop with the same name as a variable in the global scope. The global scope contains some commonly named variables, so I think it's better to ignore it. What do you think?
https://github.com/sindresorhus/globals/blob/64d801f42fd9a35617622b1b19010701352efe6d/globals.json#L683

```suggestion
            if (!variable || variable.defs.length === 0) continue
```

Reply:
Thank you for the quick feedback!

Nice catch! I even didn't think about this.
Fixed